network/discovery/dv5_service.go

```diff
@@ -152,9 +152,13 @@ func (dvs *DiscV5Service) checkPeer(logger *zap.Logger, e PeerEvent) error {
 	if err != nil {
 		return errors.Wrap(err, "could not read domain type")
 	}
-	if nodeDomainType != dvs.domainType.DomainType() &&
-		nodeDomainType != dvs.domainType.NextDomainType() {
-		return fmt.Errorf("mismatched domain type: %x", nodeDomainType)
+	nodeNextDomainType, err := records.GetDomainTypeEntry(e.Node.Record(), records.KeyNextDomainType)
+	if err != nil && !errors.Is(err, records.ErrEntryNotFound) {
+		return errors.Wrap(err, "could not read domain type")
+	}
+	if dvs.domainType.DomainType() != nodeDomainType &&
+		dvs.domainType.DomainType() != nodeNextDomainType {
+		return fmt.Errorf("mismatched domain type: %x is neither %x or %x", nodeDomainType, dvs.domainType.DomainType(), dvs.domainType.NextDomainType())
```

Comment:
just a note we should keep in mind for the future, these messages can pile up and might be hard to search. we might want to have a custom Error type where we can add fields like `want` or `have` so we can print it as {"error message", fields...}. this will make our logs much more machine readable.